kview/src/main/java/dev/ikm/komet/kview/mvvm/view/pattern/PatternFieldsController.java

```diff
@@ -107,12 +129,10 @@ private void initialize() {
         setupDragNDrop(purposeVBox, purposeStackPane, (publicId) -> {
             // check to see if a pattern > purpose was already dragged into the purpose section before saving
             // to the view model
-            if (patternFieldsViewModel.getPropertyValue(PURPOSE_ENTITY) == null) {
+            if (patternFieldsViewModel.getPropertyValue(PatternFieldsViewModel.PURPOSE_ENTITY) == null) {
```

Comment:
Is a null check needed? 

I thought if the value is not null (a previous value) the drop operation would not succeed (replacing the previous entity).

Reply:
Suggestion: If null check is needed.  
do the following:
```java
 ObjectProperty<ConceptEntity> purposeProp = patternFieldsViewModel.getProperty(PURPOSE_ENTITY);
 
if (purposeProp.isEmpty().get()) { 
    // do thing 
}
```

Reply:
updated